File: Multi_Language_CodeExamples/scala/classes.scala
/* Scala Classes */

case class Person(name: String, age: Int) {
  def greet(): String = s"Hello, I'm $name and I'm $age years old"
}

object PersonApp {
  def main(args: Array[String]): Unit = {
    val person = Person("Iris", 28)
    println(person.greet())
  }
}

File: Multi_Language_CodeExamples/lua/classes.lua
-- Lua Classes

Person = {}
Person.__index = Person

function Person:new(name, age)
    local self = setmetatable({}, Person)
    self.name = name
    self.age = age
    return self
end

function Person:greet()
    print("Hello, I'm " .. self.name .. " and I'm " .. self.age .. " years old")
end

local person = Person:new("Grace", 32)
person:greet()
